From e49e1119ed25ff1afabb7b8ee75e652900d5351b Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:21 -0400
Subject: [ethernet] bnx2x: clamp num_queues to prevent passing a negative value

Message-id: <1409946615-5862-17-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91565
O-Subject: [RHEL7.1 PATCH 16/70] bnx2x: clamp num_queues to prevent passing a negative value
Bugzilla: 1059554 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1059554
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 7d0445d66a7619e23257570daa794dbf1d89c2a6
Author: Michal Schmidt <mschmidt@redhat.com>
Date:   Tue Feb 25 16:04:24 2014 +0100

    bnx2x: clamp num_queues to prevent passing a negative value

    Use the clamp() macro to make the calculation of the number of queues
    slightly easier to understand. It also avoids a crash when someone
    accidentally passes a negative value in num_queues= module parameter.

    Signed-off-by: Michal Schmidt <mschmidt@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
index c631640..5a7bd6a 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
@@ -61,10 +61,9 @@ static void bnx2x_add_all_napi(struct bnx2x *bp)
 
 static int bnx2x_calc_num_queues(struct bnx2x *bp)
 {
- return  bnx2x_num_queues ?
-   min_t(int, bnx2x_num_queues, BNX2X_MAX_QUEUES(bp)) :
-   min_t(int, netif_get_num_default_rss_queues(),
-         BNX2X_MAX_QUEUES(bp));
+ int nq = bnx2x_num_queues ? : netif_get_num_default_rss_queues();
+ nq = clamp(nq, 1, BNX2X_MAX_QUEUES(bp));
+ return nq;
 }
 
 /**
-- 
1.7.1